### DISCUSSION TITLE
clear all state

### DISCUSSION THREAD

**Discussion by juyeong1260** [facebookexperimental/Recoil#1692] [Q&A]
Can I clear all state?

I wan't clear all recoil state when user logout. But I can't find the method for invalidate all recoil state.

I don't want to execute `window.reload()`

Please help.

**Comment by drarmstr** [comment]
You could try remounting your `<RecoilRoot>` or change its `key` prop.  Another option might be to use `useGotoRecoilSnapshot()`.  Note that selector caches are still maintained if you clear atom state, but `useRecoilRefresher_UNSTABLE()` can clear those.
